sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIOWriteExceptionHandlingTest.java

```diff
@@ -73,34 +75,36 @@ public static Collection<Object[]> data() {
     return Arrays.asList(
         new Object[][] {
           // DEADLINE_EXCEEDED is the only exception type that generates retries in-SDK
-          // The default backoff generates 9 retries, and then errors out the pipeline.
-          {ErrorCode.DEADLINE_EXCEEDED, "deadline passed!", 9, 10},
+          // The default backoff generates at least 8 retries, but can sometimes generate as many as 10 retries due to
```

Comment:
Do we want to test it retries at most 10 times?

Reply:
From the test description I'm not sure we care about anything other than DEADLINE_EXCEEDED generates > 1 attempt since the actual number of attempts depends on the settings of the backoff algorithm which could change over time. 

TBH it seems to me that the thing we really want to test is that the number of retries is one less than the number of actual calls at which point the whole approach changes

Reply:
I'm not sure what the spotless is (it's my second day ;-)) but happy to fix if you can point me in the right direction.

From your comment it also sounds like we're amenable to fixing this test up to really test that there are retries+1 calls in general with a check that retries happen when they are expected rather than fixing on specific numbers of retries? That should be a pretty straightforward change I think (well, the +1 thing might take a little work)

Reply:
I mean I agree that "generates > 1 attempt " and the current form is good to me. spotless is the check (here)[https://ci-beam.apache.org/job/beam_PreCommit_Spotless_Commit/21571/] and could be resolved by simply run `./gradlew :sdks:java:io:google-cloud-platform:spotlessApply` as hinted by the log there. Could push a commit to resolve this. btw this reminds me that may be we should config the repo such that automatically run this whenever do a push